File: totosuki/ABC/B/189.py
from decimal import Decimal
import sys
input = sys.stdin.readline

N, X = map(int, input().split())
cnt = 0
answer = -1
alcohol = 0

for i in range(N):
  V, P = map(Decimal, input().split())
  alcohol += V * (P / 100)  
  cnt += 1
  if alcohol > X:
    answer = cnt
    break

print(answer)

File: OCH_Tao/ABC/B/189.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X] = input.shift().split(" ").map(x => parseInt(x, 10));
let cnt = 0;
for (let i = 0; i < N; i++) {
  const [V, P] = input[i].split(" ").map(x => parseInt(x, 10));
  cnt += V * P;
  if (cnt > X * 100) {
    console.log(i + 1);
    return false;
  }
}
console.log(-1);